src/aux-server/aux-web/aux-player/PlayerHome/PlayerHome.ts

```diff
@@ -316,8 +330,20 @@ export default class PlayerHome extends Vue {
     async onBiosSelectionChanged() {
         if (isStaticInst(this.biosSelection)) {
             this.instOptions = await appManager.listStaticInsts();
+            this.instName = '';
+            if (!this.instOptions.includes(this.instSelection)) {
+                this.instSelection = 'new-inst';
+            }
         } else {
             this.instOptions = [];
+            this.instSelection = null;
+        }
+    }
+
+    @Watch('instSelection')
+    onInstSelectionChanged() {
+        if (this.instSelection !== 'new-inst') {
+            this.instName = '';
         }
     }
```

Comment:
@TroyceGowdy It looks like if the user changes the `instSelection` or `biosSelection`, then whatever they previously typed into the `instName` input box will be cleared. Do we have to clear the inst name when they change the selection, or would it be easier for the user to understand if their input was preserved? Would the user be more surprised if their input was cleared or if it was preserved?

There's technically nothing wrong either way, but one does appear to be better to me than the other based on the [principle of least surprise](https://en.wikipedia.org/wiki/Principle_of_least_astonishment).